docs/upgrading_binary.rst

```diff
@@ -0,0 +1,190 @@
+  that :code:`BinaryAttribute` and :code:`BinarySetAttribute` were not used in practice within maps and lists
+  before PynamoDB 6.0 and thus will not require any legacy encoding.
+
+
+Guidance
+########
+
+Top-level binary attributes
+***************************
+
+- In models existing at the time of an upgrade from PynamoDB 5 (or lower), use :code:`legacy_encoding=True`.
+
+  .. note::
+
+     In PynamoDB 6 we require this new parameter to be explicitly set to prevent inadvertent data corruption
+     during upgrades. By setting it to :code:`True` during an upgrade, the developer marks the attribute as pre-existing
+     and thus requiring legacy handling.
+
+  For example:
+
+  .. code-block:: diff
+
+      class SomeExistingModel(Model):
+     -  my_binary = BinaryAttribute()
+     +  my_binary = BinaryAttribute(legacy_encoding=True)
+
+  .. code-block:: diff
+
+      class SomeExistingModel(Model):
+     -  my_binary = BinarySetAttribute()
+     +  my_binary = BinarySetAttribute(legacy_encoding=True)
+
+  After the version upgrade is complete, you can consider adding a new binary attribute
+  and :ref:`migrating the data <migrating>`.
+
+- In new models, use :code:`legacy_encoding=False`.
+
+  .. code-block:: python
+
+     class NewModel(Model):
+       my_binary = BinaryAttribute(legacy_encoding=False)
+       my_binary_set = BinarySetAttribute(legacy_encoding=False)
+
+
+Nested binary attributes
+************************
+
+- In maps, use :code:`legacy_encoding=False`.
+
+  .. code-block:: python
+
+     class MyMap(MapAttribute):
+       binary = BinaryAttribute(legacy_encoding=False)
+       binary_set = BinarySetAttribute(legacy_encoding=False)
+
+- In raw maps, normal (non-legacy) encoding will be used.
+
+  .. code-block:: python
+
+     class MyModel(Model):
+       my_raw_map = MapAttribute()
+
+     my_model = MyModel()
+     my_model.my_raw_map = MapAttribute(binary=b'foo')
+
+- In lists, normal (non-legacy) encoding will be used.
+
+  This applies to both :code:`ListAttribute(of=BinaryAttribute)` and
+  :code:`of=BinarySetAttribute` as well as when :code:`of=...`
+  is not specified (for :code:`bytes` and :code:`Set[bytes]` elements).
+
+  For example:
+
+  .. code-block:: python
+
+     class MyModel(Model):
+       binary_list = ListAttribute(of=BinaryAttribute)
+       binary_set_list = ListAttribute(of=BinarySetAttribute)
+       mixed_list = ListAttribute()
+
+
+     model = MyModel()
+     model.binary_list = [b'\xCA', b'\xFE']
+     model.binary_set_list = [{b'\xCA', b'\xFE'}, {b'\xF0', b'\x0D'}]
+     model.mixed_list = [
+        b'\xCA\xFE',
+        {b'\xF0', b'\x0D'},
+     ]
+
+
+.. _migrating:
+
+Migrating
+#########
+
+Migrating existing systems off the legacy encoding is not necessary at this time. For large tables,
+there might be significant cost and engineering complexity involved. If you choose to do so,
+follow the typical steps for data migration:
```

Comment:
Might not hurt to include a brief warning to ensure a table backup exists before running any migrations?